File: 현세미/백준/겹치는건 싫어.py
# 백준 - 20922 겹치는 건 싫어
n, k = map(int, input().split())
arr = list(map(int, input().split()))

answer = 0
start, end = 0, 0
counter = [0] * (max(arr) + 1)

while end < n:
    if counter[arr[end]] < k:
        counter[arr[end]] += 1
        end += 1
    else:
        counter[arr[start]] -= 1
        start += 1

    answer = max(answer, end - start)

print(answer)

File: 정민선/백준/etc/겹치는건싫어.js
const fs = require('fs');
let input = fs.readFileSync('./정민선/백준/input.txt').toString().trim().split("\n");
let [n, k] = input[0].split(" ").map(Number);
let numbers = input[1].split(" ").map(Number);
let count = Array(100001).fill(0);
let start = 0;
let answer = 0;

for (let i=0; i<n; i++){
    count[numbers[i]] += 1;

    while (count[numbers[i]] > k) {
        count[numbers[start]] -= 1;
        start += 1;
    }
    answer = Math.max(answer, i-start+1);
}
console.log(answer);